pkg/vault/vault.go

```diff
@@ -106,14 +114,81 @@ func (v *Vault) ReadSecret(path string) ([]byte, error) {
 
 	data, ok := secret.Data["value"].(string)
 	if !ok {
-		return nil, fmt.Errorf("secret data type is not string, found %T value: %#v", secret.Data["value"], secret.Data["value"])
+		return nil, fmt.Errorf("secret data type is not string, found %T value: %#v at path: %s", secret.Data["value"], secret.Data["value"], path)
 	}
 
 	return []byte(data), nil
 }
 
+func (v *Vault) manageTokenLifecycle(ctx context.Context, authToken *hashivault.Secret) error {
+	authTokenWatcher, err := v.client.NewLifetimeWatcher(&hashivault.LifetimeWatcherInput{
+		Secret: v.token,
+	})
+	if err != nil {
+		return fmt.Errorf("error initializing auth token lifetime watcher: %v", err)
+	}
+
+	go authTokenWatcher.Start()
+	defer authTokenWatcher.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+
+		case <-authTokenWatcher.DoneCh():
+			// Token failed to renew (e.g expired), re-auth required
+			return nil
+
+		case renewalInfo := <-authTokenWatcher.RenewCh():
+			// Token was successfully renewed
+			if renewalInfo.Secret.Auth != nil {
+				level.Debug(v.logger).Log("msg", fmt.Sprintf("token renewed, new lease: %d", renewalInfo.Secret.Auth.LeaseDuration))
```

Comment:
I'd suggest to avoid formatting full log message:

```suggestion
				level.Debug(v.logger).Log("msg", "token renewed", "leaseDuration", renewalInfo.Secret.Auth.LeaseDuration)
```

But also, `LeaseDuration` is integer. What unit does it use? Can we format it as `time.Duration`?